src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,299 @@
+
+            }
+        });
+        let layerName = this.core.getAttribute(layerNode, 'name');
+        this.layerInfo[layerName] = layerNode;
+
+    };
+
+    ImportKeras.prototype._addConfigurableNodes = function (layerNode, layerConfig) {
+        let allPointerNames = this.core.getValidPointerNames(layerNode);
+        let config = layerConfig.config;
+        this.logger.debug(`Layer ${this.core.getAttribute(layerNode, 'name')}` +
+            ` has following configurable attributes ${allPointerNames.join(', ')}`);
+        allPointerNames.filter(pointer => !!config[pointer])
+            .forEach((pointer) => {
+                if (typeof config[pointer] == 'string') {
+                    this._addStringPropertiesNode(layerNode, config, pointer);
+                } else {
+                    this._addPluggableNodes(layerNode, config, pointer);
+                }
+            });
+    };
+
+    ImportKeras.prototype._addStringPropertiesNode = function(layerNode, config, pointer) {
+        let configurableNode = this.core.createNode({
+            parent: layerNode,
+            base: this.META[config[pointer]]
+        });
+        // This will set the necessary pointers.
+        // Of things like activations and so on...
+        this.core.setPointer(layerNode, pointer, configurableNode);
+        this.logger.debug(`Added ${this.core.getAttribute(configurableNode, 'name')}`
+            + ` as ${pointer} to the layer `
+            + `${this.core.getAttribute(layerNode, 'name')}`);
+    };
+
+    ImportKeras.prototype._addPluggableNodes = function (layerNode, config, pointer){
+        let pluggableNode = this.core.createNode({
+            parent: layerNode,
+            base: this.META[config[pointer].class_name]
+        });
+        this.logger.debug(`Added ${this.core.getAttribute(pluggableNode, 'name')} as` +
+            ` ${pointer} to the layer ${this.core.getAttribute(layerNode, 'name')}`);
+        let validArgumentsForThisNode = this.core.getValidAttributeNames(pluggableNode);
+        let configForAddedNode = config[pointer].config;
+        if (validArgumentsForThisNode && configForAddedNode) {
+            validArgumentsForThisNode.forEach((arg) => {
+                if (configForAddedNode[arg])
+                    this.core.setAttribute(pluggableNode, arg,
+                        this._toPythonIterable(configForAddedNode[arg]));
+            });
+        }
+    };
+
+    // This method is used to convert javascript arrays to a
+    // tuple/ list(Python) in string Representation. Needed for
+    // Code generation.
+    ImportKeras.prototype._toPythonIterable = function (obj) {
+        if (obj == null) {
+            return 'None';
+        }
+        if (obj instanceof Array) {
+            return '[' + obj.map((val) => {
+                return this._toPythonIterable(val);
+            }).join(', ') + ']';
+        } else {
+            return obj;
+        }
+    };
```

Comment:
> Actually, upon testing it a bit further, I found a couple bugs around parsing and setting attributes. Essentially, I imported the `redshiftModel.json` file and then generated code to ensure that it was valid keras and encountered the following error:
> 
> ```
> NameError: name 'false' not defined
> ```
I think I know the cause of the problem you alluded to, booleans start with capital case in python which I had not considered here. Refactoring this to the following should do the job.
```javascript
// This method is used to convert javascript arrays/booleans to a
    // list(python)/boolean in string Representation. Needed for
    // Code generation.
    ImportKeras.prototype._toPythonType = function (obj) {
        if (obj == null) {
            return 'None';
        }
        if (obj instanceof Array) {
            return '[' + obj.map((val) => {
                return this._toPythonType(val);
            }).join(', ') + ']';
        } else if (typeof obj === 'boolean') {
            return obj ? 'True' : 'False';
        } else {
            return obj;
        }
    };
```

Reply:
On a minor side note, it might make sense test `GenerateKeras` on this plugin as a test.

Reply:
Yep. I think it is also expecting tuples (`(1,2,3)`) rather than arrays which also caused an error after fixing the booleans. However, this is something that might be better to fix in GenerateKeras (https://github.com/deepforge-dev/deepforge-keras/blob/master/src/plugins/GenerateKeras/GenerateKeras.js#L416)